theme/src/components/GlobalTableOfContents.tsx

```diff
@@ -25,8 +25,53 @@ const HEADER_TITLES = [Header.Level2, Header.Level3];
  * data is deeply nested.
  */
 export function GlobalTableOfContents({ headers, rootHeaders }: Props) {
+  // Ref for the DOM node of the root unordered list element. This is
+  // used to get the maximum width of the container, which is used for
+  // determining if the current item overflows.
+  const listRef = useRef<HTMLOListElement>(null);
+
+  // Ref for the first DOM node that contains a list item. This node is used to
+  // measure the left border and padding to determine if the current item
+  // overflowed or not. Only one is needed because every item has the same left
+  // border and padding values.
+  const itemContainerRef = useRef<HTMLDivElement>();
+
+  /// Ref for an array of DOM nodes that correspond to a TOC item.
+  const itemsRef = useRef(new Map<string, HTMLElement>());
+
+  // Set containing list of strings that can be used to determine if a tooltip is enabled.
+  const [enabledTooltipSet, setEnabledTooltipSet] = useState(new Set<string>());
+
+  // Next.js router path.
   const currentPathname = useCurrentPathname();
 
+  // Effect that creates a new set of header IDs whose content is truncated so
+  // that we can conditionally render content tooltips.
+  useEffect(() => {
+    const nextSet = new Set<string>();
+
+    // Get left padding and border pixel values of the surrounding div container.
+    const computedStyle =
+      itemContainerRef.current &&
+      window.getComputedStyle(itemContainerRef.current);
+    const paddingLeft = parseFloat(computedStyle?.paddingLeft ?? '0');
```

Comment:
if it's pixels, can it even come in floats or only ints (doesn't really matter either way)

Reply:
Yes it can! Pixels are weird in CSS: https://stackoverflow.com/questions/20512805/does-floating-point-length-value-make-sense-on-pixel-unit

Because of this, it's possible to have floating point pixel values:
![image](https://user-images.githubusercontent.com/2176050/135940446-ba2b2694-2e79-48c3-84eb-370b72c1ec24.png)